components/transactions/TransactionRejectButton.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+
+import { getErrorFromGraphqlException } from '../../lib/errors/';
+import { GraphQLContext } from '../../lib/graphql/context';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import { Box, Flex } from '../Grid';
+import MessageBox from '../MessageBox';
+import StyledButton from '../StyledButton';
+
+/* eslint-disable graphql/template-strings */
+export const rejectTransactionMutation = gqlV2/* GraphQL */ `
+  mutation RejectTransaction($transaction: TransactionReferenceInput!) {
+    rejectTransaction(transaction: $transaction) {
+      id
+    }
+  }
+`;
+
+const TransactionRejectButton = props => {
+  const [rejectTransaction, { loading, error }] = useMutation(rejectTransactionMutation, {
+    context: API_V2_CONTEXT,
+  });
+  const transactionsDataContext = React.useContext(GraphQLContext);
+  const [isEnabled, setEnabled] = React.useState(false);
+
+  const handleRejectTransaction = async () => {
+    await rejectTransaction({ variables: { transaction: { id: props.id } } });
+    setEnabled(false);
+    transactionsDataContext?.refetch?.();
+  };
+
+  return (
+    <Flex flexDirection="column">
+      <Box>
+        {!isEnabled && (
+          <StyledButton
+            buttonSize="small"
+            minWidth={140}
+            background="transparent"
+            textTransform="capitalize"
+            onClick={() => setEnabled(true)}
+          >
+            <FormattedMessage id="actions.reject" defaultMessage="Reject" />
+          </StyledButton>
+        )}
+        {isEnabled && (
+          <React.Fragment>
+            <StyledButton
+              minWidth={140}
+              buttonSize="small"
+              mr={2}
+              textTransform="capitalize"
+              onClick={() => setEnabled(false)}
+            >
+              <FormattedMessage id="form.cancel" defaultMessage="cancel" />
```

Comment:
We have a capitalized version of this one (we should get rid of the other)

```suggestion
              onClick={() => setEnabled(false)}
            >
              <FormattedMessage id="actions.cancel" defaultMessage="Cancel" />
```